TopDrawer/Settings/Settings.swift

```diff
@@ -115,6 +115,8 @@ class Settings {
     init(keyValueStore: KeyValueStore) {
         self.path = Setting(keyValueStore: keyValueStore, key: "path", defaultValue: "")
         self.shortenPaths = Setting(keyValueStore: keyValueStore, key: "shortenPaths", defaultValue: true)
+        self.launchAtLogin = Setting(keyValueStore: keyValueStore, key: AppDelegate.launchAtLoginUserDefaultKey, defaultValue: true)
+        SMLoginItemSetEnabled(AppDelegate.launcherAppId as CFString, true)
```

Comment:
Looks like there's a bug here. When `Settings` is initialised the `SMLoginItemSetEnabled(true)` is set, which would set the application to open at login every time it's run, even if the user has disabled it?

I think the value passed to `SMLoginItemSetEnabled()` should be the current open at login state. So `SMLoginItemSetEnabled(self.launchAtLogin.value)`

Reply:
yep, true. That always made it launch on login. Actually we dont need to sync it at all as SMLoginItem are persistently stored anyways.

For me it now works fine, enabling or disabling the option.

I do compile the app, launch the .app bundle that lays in the Product folder. From there, when I log out and log in again, it works as expected. Also when disabled now:)

Reply:
I also tested it when moving to the Application folder. Works as well..